packages/mantine/src/components/table/layouts/row-layout/RowLayoutBody.tsx

```diff
@@ -1,47 +1,46 @@
+import {CompoundStylesApiProps} from '@mantine/core/lib/core/styles-api/styles-api.types';
 import {flexRender} from '@tanstack/react-table';
 import {defaultColumnSizing} from '@tanstack/table-core';
-import cx from 'clsx';
-import {Fragment, type MouseEvent} from 'react';
-import {TableLayout} from '../Table.types';
-import {useTable} from '../TableContext';
-import {TableCollapsibleColumn} from '../table-column/TableCollapsibleColumn';
-import {TableSelectableColumn} from '../table-column/TableSelectableColumn';
-import {Th} from '../table-header/Th';
-import {TableLoading} from '../table-loading/TableLoading';
-import RowLayoutClasses from './RowLayout.module.css';
-import {RowLayoutProps} from './RowLayout.types';
+import {ForwardedRef, Fragment, type MouseEvent} from 'react';
+import {CustomComponentThemeExtend, identity} from '../../../../utils';
+import {TableCollapsibleColumn} from '../../table-column/TableCollapsibleColumn';
+import {TableSelectableColumn} from '../../table-column/TableSelectableColumn';
+import {TableLoading} from '../../table-loading/TableLoading';
+import {useTable} from '../../TableContext';
+import {TableLayoutProps} from '../TableLayouts';
+import {useRowLayout} from './RowLayoutContext';
 
-const RowLayoutHeader = <T,>({table}: RowLayoutProps<T>) => {
-    const {multiRowSelectionEnabled, disableRowSelection} = useTable();
+export type RowLayoutBodyStylesNames = 'row' | 'cell' | 'collapsibleRow' | 'collapsibleWrapper';
 
-    const headers = table.getHeaderGroups().map((headerGroup) => (
-        <tr
-            key={headerGroup.id}
-            className={cx(RowLayoutClasses.headerColumns, {
-                [RowLayoutClasses.headerColumnsEnabled]: !disableRowSelection,
-                [RowLayoutClasses.headerColumnsDisabled]: disableRowSelection,
-                [RowLayoutClasses.rowSelectedMultiSelectEnabled]: multiRowSelectionEnabled,
-                [RowLayoutClasses.rowSelectedMultiSelectDisabled]: !multiRowSelectionEnabled,
-            })}
-        >
-            {headerGroup.headers.map((columnHeader) => (
-                <Th key={columnHeader.id} header={columnHeader} />
-            ))}
-        </tr>
-    ));
-    return <>{headers}</>;
-};
+export interface RowLayoutBodyProps<T>
+    extends BoxProps,
+        TableLayoutProps<T>,
+        CompoundStylesApiProps<RowLayoutBodyFactory> {}
+
+export type RowLayoutBodyFactory = Factory<{
+    props: RowLayoutBodyProps<unknown>;
+    ref: HTMLTableRowElement;
+    stylesNames: RowLayoutBodyStylesNames;
+    compound: true;
+}>;
 
-const RowLayoutBody = <T,>({
-    table,
-    doubleClickAction,
-    getExpandChildren,
-    loading,
-    keepSelection,
-    ...others
-}: RowLayoutProps<T>) => {
+const defaultProps: Partial<RowLayoutBodyProps<unknown>> = {};
+
+export const RowLayoutBody = <T,>(props: RowLayoutBodyProps<T> & {ref?: ForwardedRef<HTMLTableRowElement>}) => {
+    const ctx = useRowLayout();
```

Comment:
I did not do all my homework to find the reason why you did this. But since this hook is used only to pass style, couldn't we leverage the already defined useTableStyles already wrapping the whole _shabang_ ?

Reply:
Two folds:
1. I wanted to create a separation between the Table and the RowLayout because you can use the Table with a different layout
2. I didn't want to put both RowLayoutBody and RowLayoutHeader in the theme because they fit together and RowLayout made more sense